### ISSUE TITLE
First message published to a topic after ADM/AEM startup does not get decoded/encoded

### ISSUE THREAD

**Issue by mvs5465** [usdot-jpo-ode/asn1_codec#12]
This was reported by WyDOT and ODE team created a ticket for it [#224](https://github.com/usdot-jpo-ode/jpo-ode/issues/224). The first message that gets sent to the codec module does not get encoded/decoded, requiring it to be resent. Here's what the `log.info` file looks like when two messages are sent, only the second one gets read:

```
[180209 16:47:31.014797] [trace] Waiting for needed consumer topic: topic.Asn1EncoderInput.
[180209 16:47:31.022779] [warning] Metadata did not contain topic: topic.Asn1EncoderInput.
[180209 16:47:32.523523] [trace] Waiting for needed consumer topic: topic.Asn1EncoderInput.
[180209 16:47:32.527458] [warning] Metadata did not contain topic: topic.Asn1EncoderInput.
[180209 16:47:34.027599] [trace] Waiting for needed consumer topic: topic.Asn1EncoderInput.
[180209 16:47:34.031222] [warning] Metadata did not contain topic: topic.Asn1EncoderInput.
[180209 16:47:35.531693] [trace] Waiting for needed consumer topic: topic.Asn1EncoderInput.
[180209 16:47:35.537131] [info] Topic: topic.Asn1EncoderInput found in the kafka metadata.
[180209 16:47:35.537147] [trace] Consumer topic: topic.Asn1EncoderInput is available.
[180209 16:47:35.537276] [info] Consumer: rdkafka#consumer-1 created using topics: topic.Asn1EncoderInput.
[180209 16:47:35.537495] [info] Producer: rdkafka#producer-2 created using topic: topic.Asn1EncoderOutput.
[180209 16:47:38.693718] [trace] process_message(): starting...
[180209 16:47:38.693733] [info] ODE BSM consumer partition end of file, but ASN1_Codec still alive.
[180209 16:47:38.693738] [trace] process_message(): finished...
[180209 16:47:43.693683] [trace] process_message(): starting...
[180209 16:47:43.693736] [info] process_message(): Waiting for more BSMs from the ODE producer.
[180209 16:47:43.693765] [trace] process_message(): finished...
[180209 16:47:47.703694] [trace] process_message(): starting...
[180209 16:47:47.703721] [trace] process_message(): Read message at byte offset: 1 with length 2917
[180209 16:47:47.703745] [trace] process_message(): Message timestamp: create time, type: 1518194867677
[180209 16:47:47.718388] [trace] run(): successful encoding/decoding
[180209 16:47:47.810927] [trace] process_message(): starting...
[180209 16:47:47.810950] [info] ODE BSM consumer partition end of file, but ASN1_Codec still alive.
[180209 16:47:47.810958] [trace] process_message(): finished...
```

Seems that the consumers do not get created until the topics do, and the ODE creates topics dynamically as messages get sent through.

**Comment by jmcarter9t** [comment]
Are you able to attached the message that was sent to cause this error?  I know it is something other than the message, but just wanted to make sure I can replicate with the message.

**Comment by mvs5465** [comment]
So it happens with all messages and types, but this is a sample distress message (TIM) log file that goes to the decoder:  
[dnMsg_1506986050_2001_3A470_3A41af_3A1_3A226_3Aadff_3Afe05_3A13f1.csv.txt](https://github.com/usdot-jpo-ode/asn1_codec/files/1711989/dnMsg_1506986050_2001_3A470_3A41af_3A1_3A226_3Aadff_3Afe05_3A13f1.csv.txt)

And here is the XML of a TIM that gets sent to the encoder: 
```
<OdeAsn1Data><metadata><payloadType>us.dot.its.jpo.ode.model.OdeTimPayload</payloadType><serialId><streamId>38be72fb-5eb2-4f72-b227-b4085bd4a2ec</streamId><bundleSize>1</bundleSize><bundleId>0</bundleId><recordId>0</recordId><serialNumber>0</serialNumber></serialId><odeReceivedAt>2018-02-09T18:10:13.751Z[UTC]</odeReceivedAt><schemaVersion>4</schemaVersion><sanitized>false</sanitized><request><ode><version>2</version><index>54</index><verb>0</verb></ode><rsus><rsuTarget>127.0.0.1</rsuTarget><rsuUsername>v3user</rsuUsername><rsuPassword>fakePassword</rsuPassword><rsuRetries>2</rsuRetries><rsuTimeout>5000</rsuTimeout></rsus><snmp><rsuid>00000083</rsuid><msgid>31</msgid><mode>1</mode><channel>178</channel><interval>2</interval><deliverystart>2017-06-01T17:47:11-05:00</deliverystart><deliverystop>2018-02-02T17:47:11-05:15</deliverystop><enable>1</enable><status>4</status></snmp></request><encodings><encodings><elementName>MessageFrame</elementName><elementType>MessageFrame</elementType><encodingRule>UPER</encodingRule></encodings></encodings></metadata><payload><dataType>MessageFrame</dataType><data><MessageFrame><messageId>31</messageId><value><TravelerInformation><msgCnt>2</msgCnt><timeStamp>309505</timeStamp><packetID>EC9C236B0000000000</packetID><urlB>null</urlB><dataFrames><TravelerDataFrame><sspTimRights>0</sspTimRights><frameType><advisory></advisory></frameType><msgId><roadSignID><position><lat>416784730</lat><long>-1087827750</long><elevation>9171</elevation></position><viewAngle>1010101010101010</viewAngle><mutcdCode><warning></warning></mutcdCode><crc>0000</crc></roadSignID></msgId><startYear>2017</startYear><startTime>308065</startTime><duratonTime>1</duratonTime><priority>0</priority><sspLocationRights>3</sspLocationRights><regions><GeographicalPath><name>Testing TIM</name><id><region>0</region><id>33</id></id><anchor><lat>412500807</lat><long>-1110093847</long><elevation>20206</elevation></anchor><laneWidth>700</laneWidth><directionality><both></both></directionality><closedPath><false /></closedPath><direction>0000000000001010</direction><description><path><scale>0</scale><offset><ll><nodes><NodeLL><delta><node-LL3><lon>31024</lon><lat>14506</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>30974</lon><lat>14568</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>30983</lon><lat>14559</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>30980</lon><lat>14563</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>30982</lon><lat>14562</lat></node-LL3></delta></NodeLL></nodes></ll></offset></path></description></GeographicalPath></regions><sspMsgRights1>2</sspMsgRights1><sspMsgRights2>3</sspMsgRights2><content><advisory><SEQUENCE><item><itis>513</itis></item></SEQUENCE></advisory></content><url>null</url></TravelerDataFrame></dataFrames></TravelerInformation></value></MessageFrame></data></payload></OdeAsn1Data>
```

**Comment by aferber** [comment]
@mvs5465 I tested the latest build (and latest dependencies) but could not reproduce this using my kafka-docker setup. I don't having a working ODE build at the moment; do you think that would help with reproducibility?

**Comment by mvs5465** [comment]
A solution has been found and merged in PR #13.
